cg/cli/workflow/mip_dna/base.py

```diff
@@ -272,33 +261,124 @@ def run(
         raise click.Abort
 
 
-@mip_dna.command()
+@mip_dna.command("resolve-compression")
+@ARGUMENT_CASE_ID
 @OPTION_DRY
 @click.pass_context
-def start(context: click.Context, dry_run: bool = False):
-    """Start all cases that are ready for analysis"""
+def resolve_compression(context: click.Context, case_id: str, dry_run: bool):
+    """Handles cases where decompression is needed before starting analysis"""
     dna_api = context.obj["dna_api"]
-    exit_code = EXIT_SUCCESS
+    prepare_fastq_api = context.obj["prepare_fastq_api"]
+    case_obj = dna_api.db.family(case_id)
+    if not case_obj:
+        LOG.error("Case %s does not exist in status-db", case_id)
+        raise click.Abort
+    decompression_needed = prepare_fastq_api.is_spring_decompression_needed(case_obj.internal_id)
+    if decompression_needed:
+        decompression_all_samples_started = prepare_fastq_api.start_spring_decompression(
+            case_obj.internal_id, dry_run
+        )
+        if decompression_all_samples_started:
+            LOG.info(
+                "The analysis for %s could not start, started decompression instead",
+                case_obj.internal_id,
+            )
+        else:
+            LOG.warning(
+                "The analysis for %s could not start",
+                case_obj.internal_id,
+            )
+            LOG.warning(
+                "The decompression for %s could not start",
+                case_obj.internal_id,
+            )
+        raise DecompressionNeededError("Workflow interrupted: decompression is not finished")
```

Comment:
Same thing communicated twice?
```suggestion
        LOG.warning(
                "The analysis for %s could not start",
                case_obj.internal_id,
            )
        if decompression_all_samples_started:
            LOG.info(
                "Decompression started for %s",
                case_obj.internal_id,
            )
        else:
            LOG.warning(
                "The decompression for %s could not start",
                case_obj.internal_id,
            )
        raise DecompressionNeededError("Workflow interrupted: decompression is not finished")
```